src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/services/staff/BulkPrintStaffNotificationServiceTest.java

```diff
@@ -0,0 +1,120 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class BulkPrintStaffNotificationServiceTest extends MockSpringTest {
+
+    private static final byte[] PDF_CONTENT = {1, 2, 3, 4};
+
+    @Autowired
+    private BulkPrintStaffNotificationService service;
+
+    @Captor
+    private ArgumentCaptor<String> senderArgument;
+    @Captor
+    private ArgumentCaptor<EmailData> emailDataArgument;
+
+    @Autowired
+    private StaffEmailProperties emailProperties;
+
+    private List<Document> documents;
+
+    private Claim claim;
+
+    @Before
+    public void setup() {
+        documents = new ArrayList<>();
+        documents.add(new Document("template1", new HashMap<>()));
+        documents.add(new Document("template2", new HashMap<>()));
+
+        claim = SampleClaim
+            .builder()
+            .build();
+
+        when(pdfServiceClient.generateFromHtml(any(byte[].class), anyMap()))
+            .thenReturn(PDF_CONTENT);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerWhenGivenNullClaim() {
+        service.notifyFailedBulkPrint(documents, null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void shouldThrowNullPointerWhenGivenNullDocuments() {
+        service.notifyFailedBulkPrint(null, claim);
+    }
+
+    @Test
+    public void shouldSendEmailToExpectedRecipient() {
+        service.notifyFailedBulkPrint(documents, claim);
+
+        verify(emailService).sendEmail(senderArgument.capture(), emailDataArgument.capture());
+
+        assertThat(senderArgument.getValue()).isEqualTo(emailProperties.getSender());
+    }
+
+    @Test
+    public void shouldSendEmailWithExpectedContent() {
+        service.notifyFailedBulkPrint(documents, claim);
+
+        verify(emailService).sendEmail(senderArgument.capture(), emailDataArgument.capture());
+
+        assertThat(emailDataArgument.getValue()
+            .getSubject()).startsWith("Print for claim 000CM001 failed");
+        assertThat(emailDataArgument.getValue()
+            .getMessage()).startsWith(
+            "The bulk print for this claim failed, please print and post the pin letter and claim form to the defendant"
+        );
+    }
+
+    @Test
+    public void shouldSendEmailWithExpectedPDFAttachments() throws IOException {
+        service.notifyFailedBulkPrint(documents, claim);
+
+        verify(emailService).sendEmail(senderArgument.capture(), emailDataArgument.capture());
+
+        EmailAttachment emailAttachment = emailDataArgument.getValue()
+            .getAttachments()
+            .get(0);
+
+        String expectedPinLetterFileName
```

Comment:
That test does not guarantee that in real life first document will be a PIN letter and second one will be sealed claim copy. 

I can change `publisher.publishEvent(new DocumentReadyToPrintEvent(event.getClaim(), defendantLetterDocument , sealedClaimDocument));` to be` publisher.publishEvent(new DocumentReadyToPrintEvent(event.getClaim(), sealedClaimDocument, defendantLetterDocument));` and tests will keep passing while in application will start sending PIN letters as sealed claims and vice versa.

Reply:
BulkPrintServiceTest checks for the order of documents and will fail if order changes

Reply:
No it does not, it checks only that the order is not changed within the `BulkPrintServiceTest`